akka-actor-typed/src/main/scala/akka/actor/typed/ActorRef.scala

```diff
@@ -113,3 +114,27 @@ private[akka] final case class SerializedActorRef[T] private (address: String) {
       resolver.resolveActorRef(address)
   }
 }
+
+/**
+ * FIXME doc
+ * - not serializable
+ * - not watchable
+ */
+trait MessageChannel[-T] { this: InternalMessageChannel[T] ⇒
```

Comment:
I don't like the name much, for me a channel sounds like something much more heavy weight (I think in the EIP book it is about separate programs?).

What about `Recipient` (_One that receives or is receptive._)? Or as it does not imply serialization and reference transparency `LocalRecipient`?

Reply:
I like `Recipient`. We shouldn't confuse with Local. What is local? The target actor might not be local.

Reply:
`RecipientRef` or `MessageReceiverRef`?

Reply:
that's also good names, yes it might be good to add the `Ref` for this also
`ActorRef`, `EntityRef`, `SourceRef`, `SinkRef`

Reply:
Agree with Johan, glad I didn't miss this PR :-) 
`RecipientRef` is excellent I think. Keeping `Ref` in the name is a good idea, all those things symbolize only a handle to the thing, not the thing itself so the Ref feels right here :)

Reply:
`RecipientRef` it is then, thanks